### ISSUE TITLE
Ubuntu 19.10 experience

### ISSUE THREAD

**Issue by Envoyfromhell** [SirToffski/WireGuard-Ligase#8]
Hi,

here just some problems/stuff i experienced with Ubuntu 19.10.
I am primarily a Windows user and perhaps some of the points i am writing here could be due to my inexperience with *nix-based os systems, please excuse and ignore these points.
I worked around the problems with commenting out the if and the else branch or installing the packages manually, just for your information.

- Ubuntu 19.10 can install Wireguard without the need to add the repository
- The generate server key question didn´t work for me, everytime I inputted the 1 the script did execute the else path
- The setup-firewall-question had the same problem
- The `apt-get install iptables-persistent
systemctl enable netfilter-persistent
netfilter-persistent save` block threw an error with no installation (sorry, didn´t notate the error)
- just a suggestion for the qr-codes, I would have liked it if it generate the qr code additionally as an png i can send per mail or otherwise (`qrencode -t png -o "$my_wgl_folder"/client_configs/"${client_name_["$q"]}".png <"$my_wgl_folder"/client_configs/"${client_name_["$q"]}".conf`

**Comment by SirToffski** [comment]
Thanks for the feedback. I’ll test a fresh Ubuntu 19.10 install tomorrow to try reproducing the errors.

From your feedback I presume you chose the “normal” mode, not the “quick mode”?

More updates to follow tomorrow.

Cheers

**Comment by Envoyfromhell** [comment]
Correct, i used the "Normal Setup"-Option to initially setup the server on a fresh and clean install of Ubuntu 19.10. Due to the fact that the main script didn´t start the firewall installation, i started this afterwards from the "main menu".

**Comment by SirToffski** [comment]
I've setup a new instance of Ubuntu v19.10 on a public VPS to test.

- Script has been updated to not add a PPA to Ubuntu v19.10 as per your suggestion. Confirmed it worked on my test machine. See commit [3806bd8](https://github.com/SirToffski/WireGuard-Ligase/commit/3806bd8c6f1c26aa2c02f48a120884629eec7fb9) for more info.
- Regarding generating server keys and firewall -  I was unable to reproduce your error. The script behaved as expected on my machine.

Choosing to generate server keys example below. Do note no additional user input is needed when choosing option "1". Server key will be generated in `"$my_wgl_folder"/keys/ServerPublickey`. Afterwards the script will move on to naming your WireGuard interface:

![msedge_rj2YBLbpvb](https://user-images.githubusercontent.com/47118864/80283191-32265c00-86ec-11ea-98af-26fd387558a6.png)

Choosing to configure firewall worked as well. The script has a built-in logic to extract previously defined data (e.g. _server private IP, listen port, WAN interface_, etc) to avoid repetitive questions. Should it not find the data, you will be asked the questions again. This is how it looks when choosing to configure at the end of the `normal setup` mode:

![loyDWUDbRR](https://user-images.githubusercontent.com/47118864/80283299-e58f5080-86ec-11ea-836e-9d9dfa32f953.png)

![msedge_g0U62ML9me](https://user-images.githubusercontent.com/47118864/80283308-f049e580-86ec-11ea-8895-0e9e472fb6aa.png)


Finally, I did not experience an error with:
```
apt-get install iptables-persistent 
systemctl enable netfilter-persistent 
netfilter-persistent save
```

The installation was completed normally and I was able to save my IPv4/IPv6 rules.


If you are willing to help troubleshooting, please re-run the script using `-x` option:
```
$ sudo bash -x configure-wireguard.sh
```
Doing so would run the script in a verbose mode and might provide some clues as to what is causing the unexpected behavior.

PS. I will implement the option to save QR codes shortly.

**Comment by Envoyfromhell** [comment]
Hi, I have debugged it again in a new clean VM, and tried it 5 times to be sure, and i think i found what is causing the problem, but i don´t have an solution.

If I press the arrow up key when an "Press any key" choice happens, the answer for the next question doesn´t get recognized correctly. If i used the enter key instead it worked as intended.

![Any Key Choice](https://user-images.githubusercontent.com/63925717/80289386-51e67000-873e-11ea-955a-624a6d161803.PNG)

![Private Key](https://user-images.githubusercontent.com/63925717/80289426-a25dcd80-873e-11ea-8a83-eaa23fe3571d.PNG)

![Firewall](https://user-images.githubusercontent.com/63925717/80289445-c0c3c900-873e-11ea-865a-b147e86f201e.PNG)

![Netfilter-persistent](https://user-images.githubusercontent.com/63925717/80289454-cd482180-873e-11ea-8b62-65118260271b.PNG)

The change with repository worked flawless.
And sorry for typos, english isn´t my native language ;-)

**Comment by SirToffski** [comment]
Hey thanks! Very interesting, I wouldn't have had the "up-arrow" usage in mind at all, a good find for sure.

Perhaps making prompt less ambiguous can be a good way to avoid these types of unexpected behaviors. 

I expect the QR code > file.png feature will be implemented within next couple of days as well.

Cheers!
